### PR TITLE
Feat/more bar label position

### LINKED ISSUE #1144: [Feature] bar label support position like 'top-right', 'bottom-right' , etc.
### What problem does this feature solve?

image

### What does the proposed API look like?

```js
label: {
 position: ' top-right'
}
```

### PR DESCRIPTION
**Pull request by xiaoluoHe**
[[中文版模板 / Chinese template](https://github.com/VisActor/VChart/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Release
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close #1144 

### 🐞 Bugserver case id
65407471f95bdcf7c340d6bb

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: docs/assets/option/en/series/area.md [hunk 1]
```diff
@@ -132,3 +145,7 @@ Total label, working when the data is stacked. Supported since version `1.3.0`.
   defaultOffset = 5,
   ignoreCustom = true
 ) }}
+
+```
```
Comment (kkxxkk2019) [inline docs/assets/option/en/series/area.md:149]: 这里咋额外多了两个 ```

File: docs/assets/option/en/series/line.md [hunk 2]
```diff
@@ -98,3 +110,7 @@ Connection method for non-compliant data points. Connection methods for null, un
 - 'link' means to ignore that point and keep connecting continuously
 - 'zero' means the default value of that point is 0
 - 'ignore' means not to process
+
+```
```
Comment (kkxxkk2019) [inline docs/assets/option/en/series/line.md:114]: 这里也是

File: docs/assets/option/zh/series/area.md [hunk 3]
```diff
@@ -132,3 +144,4 @@ point 图元样式配置。
   defaultOffset = 5,
   ignoreCustom = true
 ) }}
+```
```
Comment (kkxxkk2019) [inline docs/assets/option/zh/series/area.md:147]: 额外的？

File: packages/vchart/src/component/label/util.ts [hunk 4]
```diff
@@ -86,31 +96,31 @@ export function barLabel(labelInfo: ILabelInfo) {
...
   const isInverse =
     (series as ICartesianSeries).direction === 'horizontal'
       ? (series as ICartesianSeries).getXAxisHelper()?.isInverse()
       : (series as ICartesianSeries).getYAxisHelper()?.isInverse();
 
-  let position = labelPosition as BaseLabelAttrs['position'];
+  let position = originPosition as BaseLabelAttrs['position'];
 
-  if (position !== 'inside') {
+  if (isString(originPosition) && position !== 'inside') {
```
Comment (kkxxkk2019) [inline packages/vchart/src/component/label/util.ts:108]: 为啥要判断 isString

File: packages/vchart/src/series/bar/interface.ts [hunk 5]
```diff
@@ -35,9 +36,12 @@ export interface IBarSeriesSpec
    */
   [SeriesMarkNameEnum.barBackground]?: IMarkSpec<IRectMarkSpec>;
   /** 标签配置*/
-  [SeriesMarkNameEnum.label]?: ILabelSpec & {
-    /** 标签位置 */
-    position?:
+  [SeriesMarkNameEnum.label]?: Omit<ILabelSpec, 'position'> & {
+    /**
+     * 标签位置
+     * @since 1.6.0，支持'top-right' | 'top-left'| 'bottom-right' | 'bottom-left'，以及函数形式配置
```
Comment (kkxxkk2019) [inline packages/vchart/src/series/bar/interface.ts:42]: 后面的文字加到上面注释把，否则不好解析~

File: packages/vchart/src/series/line/interface.ts [hunk 6]
```diff
@@ -33,18 +34,13 @@ export interface ILineSeriesSpec
...
-      | 'left'
-      | 'right'
-      | 'top-right'
-      | 'top-left'
-      | 'bottom-right'
-      | 'bottom-left'
-      | 'center';
+  [SeriesMarkNameEnum.label]?: Omit<ILabelSpec, 'position'> & {
+    /** 标签位置
+     * @since 1.6.0，支持以函数形式配置
```
Comment (kkxxkk2019) [inline packages/vchart/src/series/line/interface.ts:39]: 这里也是

File: packages/vchart/src/series/scatter/interface.ts [hunk 7]
```diff
@@ -39,18 +40,13 @@ export interface IScatterSeriesSpec
...
-      | 'center';
+  [SeriesMarkNameEnum.label]?: Omit<ILabelSpec, 'position'> & {
+    /** 标签位置
+     * @since 1.6.0，支持以函数形式配置
+     */
+    position?: Functional<
+      'top' | 'bottom' | 'left' | 'right' | 'top-right' | 'top-left' | 'bottom-right' | 'bottom-left' | 'center'
+    >;
   };
 }
```
Comment (kkxxkk2019) [inline packages/vchart/src/series/scatter/interface.ts:52]: compile 报错了 

https://github.com/VisActor/VChart/actions/runs/6691010394/job/18177410046?pr=1369

**Comment by kkxxkk2019** [comment]
bugserver case 补充下

File: docs/assets/option/en/series/area.md (hunk 1 above)
Comment (xiaoluoHe) [reply docs/assets/option/en/series/area.md:149]: 可能是自动格式化引入的，我删一下

File: packages/vchart/src/component/label/util.ts (hunk 4 above)
Comment (xiaoluoHe) [reply packages/vchart/src/component/label/util.ts:108]: position 可能配置成函数

Comment (kkxxkk2019) [reply packages/vchart/src/component/label/util.ts:116]: 哦哦哦，uniformLabelPosition 返回的是函数

**Pull request merged**